runtime/src/append_vec.rs

```diff
@@ -453,6 +487,39 @@ pub mod tests {
         }
     }
 
+    impl<'a> StoredAccount<'a> {
+        fn set_data_len_unsafe(&self, new_data_len: u64) {
+            let data_len: &u64 = &self.meta.data_len;
+            #[allow(mutable_transmutes)]
+            // UNSAFE: cast away & (= const ref) to &mut to force to mutate append-only (=read-only) AppendVec
+            let data_len: &mut u64 = unsafe { &mut *(data_len as *const u64 as *mut u64) };
+            *data_len = new_data_len;
+        }
+
+        fn ref_executable_byte(&self) -> &u8 {
+            let executable_bool: &bool = &self.account_meta.executable;
+            // UNSAFE: Force to interpret mmap-backed bool as u8 to really read the actual memory content
+            let executable_byte: &u8 = unsafe { &*(executable_bool as *const bool as *const u8) };
```

Comment:
Can we also use this for `AppendVec::sanitize`

Reply:
Really nice! 581ae43b1e1584c55f54b90da31ddde40c91d95b